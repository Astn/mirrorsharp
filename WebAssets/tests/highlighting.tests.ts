import { TestDriver } from './test-driver';

test('C# highlighting is rendered correctly', async () => {
    if (TestDriver.shouldSkipRender)
        return;
    const driver = await TestDriver.new({ text: [
        'public class C<T> {',
        '    // test comment',
        '    public void M<T>() {',
        '        double d = 1.2e3;',
        '        string s1 = "test";',
        '        string s2 = $"a{s1}b";',
        "        char c = 't';",
        '        Action a = () => {};',
        '    }',
        '}'
    ].join('\n') });
    await driver.completeBackgroundWork();

    const rendered = await driver.render();

    expect(rendered).toMatchImageSnapshot();
});