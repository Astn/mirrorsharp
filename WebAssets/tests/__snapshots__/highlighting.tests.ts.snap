// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`C# highlighting applies expected classes 1`] = `
"<div class=\\"cm-line\\"><span class=\\"cmt-keyword\\">public</span> <span class=\\"cmt-keyword\\">class</span> C<span class=\\"cmt-punctuation\\">&lt;</span>T<span class=\\"cmt-punctuation\\">&gt;</span> <span class=\\"cmt-punctuation\\">{</span>
    <span class=\\"cmt-comment\\">// test comment</span>
    <span class=\\"cmt-keyword\\">public</span> <span class=\\"cmt-keyword\\">void</span> M<span class=\\"cmt-punctuation\\">&lt;</span>T<span class=\\"cmt-punctuation\\">&gt;</span><span class=\\"cmt-punctuation\\">(</span><span class=\\"cmt-punctuation\\">)</span> <span class=\\"cmt-punctuation\\">{</span>
        <span class=\\"cmt-keyword\\">double</span> d <span class=\\"cmt-punctuation\\">=</span> <span class=\\"cmt-number\\">1.2e3</span><span class=\\"cmt-punctuation\\">;</span>
        <span class=\\"cmt-keyword\\">string</span> s1 <span class=\\"cmt-punctuation\\">=</span> <span class=\\"cmt-string\\">\\"test\\"</span><span class=\\"cmt-punctuation\\">;</span>
        <span class=\\"cmt-keyword\\">string</span> s2 <span class=\\"cmt-punctuation\\">=</span> $<span class=\\"cmt-string\\">\\"a{s1}b\\"</span><span class=\\"cmt-punctuation\\">;</span>
        <span class=\\"cmt-keyword\\">char</span> c <span class=\\"cmt-punctuation\\">=</span> 't'<span class=\\"cmt-punctuation\\">;</span>
        Action a <span class=\\"cmt-punctuation\\">=</span> <span class=\\"cmt-punctuation\\">(</span><span class=\\"cmt-punctuation\\">)</span> <span class=\\"cmt-punctuation\\">=&gt;</span> <span class=\\"cmt-punctuation\\">{</span><span class=\\"cmt-punctuation\\">}</span><span class=\\"cmt-punctuation\\">;</span>
    <span class=\\"cmt-punctuation\\">}</span>
<span class=\\"cmt-punctuation\\">}</span></div>"
`;
